### DISCUSSION TITLE
uvicorn adding its own server header to response

### DISCUSSION THREAD

**Discussion by udit-pandey** [Kludex/uvicorn#1435] [Potential Issue] [answered]
I have added few headers using [secure package](https://pypi.org/project/secure/) in my fastapi application. I wanted to overwrite the server header(having default value "uvicorn") to something else. All the added headers using secure package are replicated in the responses except the server header which is coming twice(once with value given by me and other with uvicorn) as shown the postman api response below:

![image](https://user-images.githubusercontent.com/18224381/161194376-1ff92c6c-ed3e-41c6-9578-39993ea17179.png)

I run my application using:
`gunicorn -k uvicorn.workers.UvicornWorker ${APP_MODULE} --bind 0.0.0.0:80`

Why is this header being added again by **uvicorn** even though it already exists?

**Comment by Kludex** [comment]
https://www.uvicorn.org/settings/#http

**Reply by udit-pandey** [reply]
I thought the CLI command would clearly indicate that I am currently using uvicorn workers and gunicorn server to run my application. These settings would have been useful if I was using uvicorn to run my application.
Am I missing something? Please point out where in the document should I look. Thanks.

**Reply by Kludex** [reply]
Ah... I didn't see you were using gunicorn there. 🤔

I'm not sure how to do it... I'm going to check.

**Reply by Kludex** [reply]
Hmm... We add `date` and `server` headers even if they are already present. 🤔

**Reply by udit-pandey** [reply]
Is there any workaround for this?
Should I create an issue?

**Reply by Kludex** [reply]
I've automatically created #1436. If you can add more information there, it would be cool.

**Comment by Kludex** [comment, marked as answer]
Copied from my own answer on https://github.com/encode/uvicorn/issues/1436#issuecomment-1127228232:

The solution here, as many analogous (considering the arguments that are not passed to Uvicorn from Gunicorn), is to subclass `UvicornWorker`.

Example:
```python
from uvicorn.workers import UvicornWorker

class HeadlessUvicornWorker(UvicornWorker):
    CONFIG_KWARGS = {"loop": "auto", "http": "auto", "date_header": False}
```
Then you can use it as, e.g.:
```bash
gunicorn main:app --worker-class=main.HeadlessUvicornWorker --bind=localhost:8001
```
